### PR TITLE
purple-facebook: init at 66ee77378d82

### PR DESCRIPTION
**Pull request by davorb**
###### Motivation for this change

Adds a package that didn't exist before.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchurl, pidgin, glib, json_glib } :
+
+stdenv.mkDerivation rec {
+  name = "purple-facebook-${version}";
+  version = "66ee77378d82";
+
+  src = fetchurl {
+    url = "https://github.com/dequis/purple-facebook/releases/download/66ee77378d82/purple-facebook-${version}.tar.gz";
```
Comment (joachifm) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: Looks like you can re-use `version` once more here.

Comment (davorb) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: Nice catch :+1:

**Comment by joachifm** [comment]
Please squash when you're ready to merge.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchurl, pidgin, glib, json_glib } :
+
+stdenv.mkDerivation rec {
+  name = "purple-facebook-${version}";
+  version = "66ee77378d82";
```
Comment (primeos) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: In the case of a commit version should be the date of that commit i.e. probably release in this case(?):

> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23". - [nixpkgs manual - 10.2. Package naming](https://nixos.org/nixpkgs/manual/#sec-package-naming)

The problem is that these release versions could start with a character while the manual states:

> The version part of the name attribute must start with a digit (following a dash) — e.g., "hello-0.3.1rc2".

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/dequis/purple-facebook/releases/download/${version}/purple-facebook-${version}.tar.gz";
```
Comment (primeos) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: Wouldn't it be possible to use `fetchFromGitHub` instead?

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  postInstall =  ''
+    mkdir -p $out/lib/purple-2
+    cp pidgin/libpurple/protocols/facebook/.libs/*.so $out/lib/purple-2/
+  '';
+
+  buildInputs = [ pidgin glib json_glib ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/dequis/purple-facebook;
```
Comment (primeos) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: If you use `fetchFromGitHub` you could use `inherit (src.meta) homepage;` here :wink:

**Review by primeos — CHANGES_REQUESTED** [review]

**Comment by davorb** [comment]
@primeos I pushed the changes you suggested with 54e0d1a, but they don't seem to be visible when I click the "files changed" tab. Do I need to do anything else?

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix (hunk 2 above)
Comment (joachifm) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: But do note that this *is* a release version, so the clause "if the package is not a release but a commit" doesn't apply. The version parsing issue remains, of course.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix (hunk 4 above)
Comment (davorb) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-facebook/default.nix]: Oh, I forgot to add this in 54e0d1a!

**Comment by primeos** [comment]
@davorb that's strange, I can see the changes - are they still not visible on your view?
The `homepage` (`inherit (src.meta) homepage;`) attribute seems to be missing though.

The rest should be fine but I don't know what to do about the `version` attribute...

**Comment by davorb** [comment]
@primeos I added the homepage thing to 7cf6f46. 

I think that the easiest thing for now might be to keep the date as the version number. That way, it will at least be easy for people to know how old their version is.

**Comment by davorb** [comment]
@joachifm I think that this should be ready to merge.

**Comment by Mic92** [comment]
Thanks!

**Comment by joachifm** [comment]
@davorb the hydra build fails, see https://hydra.nixos.org/build/41952144/log/raw

**Comment by davorb** [comment]
@joachifm Yes, I noticed. I'll send in another commit tomorrow to fix this.

**Comment by Mic92** [comment]
Where does this patch coming from? I tested this plugin with pidgin actually.

**Comment by davorb** [comment]
@Mic92 I think what's happening is that ```./autogen.sh``` is failing, and that's why you're seeing the patch-messages. I have no idea where they are coming from

I just tried rebuilding it, and while the autoreconfPhase took a surprisingly long time (minutes), it *did* build on my system. I'll try building it on a fresh install of NixOS on a virtual machine later today, and see if that changes anything. In any case, I'm reasonably sure that it's autogen that's causing all of these issues.

**Pull request merged**